runner/app/routes/segment_anything_2.py

```
@@ -0,0 +1,110 @@
+                headers={"WWW-Authenticate": "Bearer"},
+                content=http_error("Invalid bearer token"),
+            )
+
+    if model_id != "" and model_id != pipeline.model_id:
+        return JSONResponse(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            content=http_error(
+                f"pipeline configured with {pipeline.model_id} but called with "
+                f"{model_id}"
+            ),
+        )
+
+    try:
+        point_coords = json_str_to_np_array(point_coords, var_name="point_coords")
+        point_labels = json_str_to_np_array(point_labels, var_name="point_labels")
+        box = json_str_to_np_array(box, var_name="box")
+        mask_input = json_str_to_np_array(mask_input, var_name="mask_input")
+    except ValueError as e:
+        return JSONResponse(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            content=http_error(str(e)),
+        )
+
+    try:
+        image = Image.open(image.file)
+        masks, iou_predictions, low_res_masks = pipeline(
+            image,
+            point_coords=point_coords,
+            point_labels=point_labels,
+            box=box,
+            mask_input=mask_input,
+            multimask_output=multimask_output,
+            return_logits=return_logits,
+            normalize_coords=normalize_coords,
+        )
+    except Exception as e:
+        logger.error(f"SAM2 error: {e}")
+        logger.exception(e)
+        if isinstance(e, InferenceError):
+            return JSONResponse(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                content=http_error(str(e)),
+            )
+
+        return JSONResponse(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            content=http_error("SAM2 error"),
+        )
+
+    return {
+        "masks": json.dumps(masks.tolist()),
+        "iou_predictions": json.dumps(iou_predictions.tolist()),
+        "low_res_masks": json.dumps(low_res_masks.tolist()),
```

@pschroedl, what do you think about adding a boolean parameter like `low_res` to control the quality of the mask returned? This could help reduce data transfer when high resolution isn't necessary. Then if set we return the low_res_mask instead of the masks 🤔?

---

As requested by users, we're going to return all three by default, but renaming [predictions -> scores, mask -> logits](https://github.com/livepeer/ai-worker/pull/154/commits/b200ab6aaf98310855d4e7013452b3b3a2f49b08) to follow expected convention